src/server/scripts/Spells/spell_druid.cpp

```diff
@@ -1324,6 +1325,31 @@ class spell_dru_luxuriant_soil : public AuraScript
     }
 };
 
+// 33917 - Mangle
+class spell_dru_mangle : public SpellScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_DRUID_MANGLE_TALENT });
+    }
+
+    bool Load() override
+    {
+        return GetCaster()->HasAura(SPELL_DRUID_MANGLE_TALENT);
+    }
+
+    void CalculateDamage(SpellEffectInfo const& /*spellEffectInfo*/, Unit* victim, int32& /*damage*/, int32& /*flatMod*/, float& pctMod) const
+    {
+        if (victim->HasAuraState(AURA_STATE_BLEED))
+            AddPct(pctMod, GetSpellInfo()->GetEffect(EFFECT_2).CalcValue());
```

Comment:
you can just use `GetEffectInfo(EFFECT_2)` instead of `GetSpellInfo()->GetEffect(EFFECT_2)` and we gotta take the potential caster buffs into account, so we gotta add GetCaster() to CalcValue(): `CalcValue(GetCaster())`